src/js/api/set-config.js

```diff
@@ -33,6 +35,18 @@ export default (controller, newConfig) => {
         controller._view.resize(newConfig.width, newConfig.height);
     }
 
+    if (newConfig.floating) {
+        const currFloatCfg = model.get('floating') || {};
+        const currFloatMode = currFloatCfg.mode;
+        const newFloatCfg = Object.assign({}, currFloatCfg, newConfig.floating);
+        if (currFloatMode === newFloatCfg.mode || FLOAT_ENUM.indexOf(newFloatCfg.mode) === -1) {
+            delete newConfig.floating;
+        } else {
+            newConfig.floating = newFloatCfg;
+            model.set('floating', newFloatCfg);
```

Comment:
Won't `model.set(field, newValue)` handle this below?

Reply:
great catch